tlatools/src/tla2sany/semantic/ThmOrAssumpDefNode.java

```diff
@@ -312,8 +311,8 @@ public boolean addLabel(LabelNode odn) {
     * as odn, then odn is added to the set and true is return; else the    *
     * set is unchanged and false is returned.                              *
     ***********************************************************************/
-    if (labels == null) {labels = new Hashtable(); } ;
-    if (labels.containsKey(odn)) {return false ;} ;
+    if (labels == null) {labels = new Hashtable<>(); } ;
+    if (labels.containsKey(odn.getName())) {return false ;} ;
```

Comment:
This is a behavioral change. Please leave this out or write a unit test.

Reply:
By behavioral change, are you talking about the bug fix?

Reply:
Lets not bury a bugfix inside a large refactoring.

Reply:
Hmmm... ok.  The minor amount of typing added to the code is indeed only there because it was surrounding the code with the bug fix.

I can make a branch containing that single line (317) changed if that's more palatable. ?

Should i dump the other code changes all together, since there is no Issue related to the general "modernize the code with typing and annotations" (dovetailing back into the original quandary of many PRs or fewer blobby PRs.)

Reply:
Splitting the commit into two is just fine.

I got into the habit of tagging commits with [Bug], [Refactor], [Feature]... This scheme proved itself helpful when one has to bisect history to find a bug. That's why I want to split it in two to be able to safely skip [Refactor] commits.

Reply:
Sounds like a plan - will tidy up.